indexing-service/src/main/java/io/druid/indexing/overlord/HeapMemoryTaskStorage.java

```diff
@@ -198,6 +197,30 @@ public void addLock(final String taskid, final TaskLock taskLock)
   }
 
   @Override
+  public void upgradeLock(String taskid, TaskLock taskLock)
```

Comment:
I can add the part to throw exception if the caller asks for downgrading the lock from `upgradeLock` method